test/functional-test/test_userprofile.py

```diff
@@ -0,0 +1,112 @@
+import unittest
+from lib.analysis import network
+from lib.in_out import reader , saver 
+from lib.nickTracker import nick_tracker
+import networkx as nx
+import lib.util as util
+import lib.config as config
+import os
+import StringIO
+import sys
+from networkx.algorithms.components.connected import connected_components
+from numpy.testing import assert_array_equal
+
+class UserTrackingFunctionalTest(unittest.TestCase):
```

Comment:
As mentioned in issue #298, we only need two functional tests here. One for user to channel tracking and another is to user nick change graph. The test functions of network module do not belong here.